pkg/util/deployer/deployer.go

```diff
@@ -0,0 +1,125 @@
+package deployer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/fs"
+	"strings"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/ready"
+)
+
+type Deployer interface {
+	CreateOrUpdate(context.Context, *arov1alpha1.Cluster, interface{}) error
+	Remove(context.Context, interface{}) error
+	IsReady(context.Context, string, string) (bool, error)
+	Template(interface{}, fs.FS) ([]kruntime.Object, error)
+}
+
+type deployer struct {
+	kubernetescli kubernetes.Interface
+	dh            dynamichelper.Interface
+	fs            fs.FS
+}
+
+func NewDeployer(kubernetescli kubernetes.Interface, dh dynamichelper.Interface, fs fs.FS) Deployer {
+	return &deployer{
+		kubernetescli: kubernetescli,
+		dh:            dh,
+		fs:            fs,
+	}
+}
+
+func (depl *deployer) Template(data interface{}, fsys fs.FS) ([]kruntime.Object, error) {
+	results := make([]kruntime.Object, 0)
+	template, err := template.ParseFS(fsys, "staticresources/*")
```

Comment:
What if the `staticresources` are placed elsewhere by the controller? I understand this is the controllers choice?